server/lyft/gateway/autoplan_builder.go

```diff
@@ -0,0 +1,166 @@
+type AutoplanValidator interface {
+	PullRequestHasTerraformChanges(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool
+}
+
+// AutoplanBuilder handles setting up repo cloning and checking to verify of any terraform files have changed
+type AutoplanBuilder struct {
+	Logger                        logging.SimpleLogging
+	Scope                         tally.Scope
+	VCSClient                     vcs.Client
+	PreWorkflowHooksCommandRunner events.PreWorkflowHooksCommandRunner
+	DisableAutoplan               bool
+	Drainer                       *events.Drainer
+	GlobalCfg                     valid.GlobalCfg
+	// AllowForkPRs controls whether we operate on pull requests from forks.
+	AllowForkPRs bool
+	// AllowForkPRsFlag is the name of the flag that controls fork PR's. We use
+	// this in our error message back to the user on a forked PR so they know
+	// how to enable this functionality.
+	AllowForkPRsFlag string
+	// SilenceForkPRErrors controls whether to comment on Fork PRs when AllowForkPRs = False
+	SilenceForkPRErrors bool
+	// SilenceForkPRErrorsFlag is the name of the flag that controls fork PR's. We use
+	// this in our error message back to the user on a forked PR so they know
+	// how to disable error comment
+	SilenceForkPRErrorsFlag string
+	// SilenceVCSStatusNoPlans is whether autoplan should set commit status if no plans
+	// are found
+	silenceVCSStatusNoPlans bool
+	// SilenceVCSStatusNoPlans is whether any plan should set commit status if no projects
+	// are found
+	silenceVCSStatusNoProjects bool
+	CommitStatusUpdater        events.CommitStatusUpdater
+	PrjCmdBuilder              events.ProjectPlanCommandBuilder
+	PullUpdater                *events.PullUpdater
+}
+
+func (r *AutoplanBuilder) PullRequestHasTerraformChanges(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool {
+	if opStarted := r.Drainer.StartOp(); !opStarted {
+		if commentErr := r.VCSClient.CreateComment(baseRepo, pull.Num, ShutdownComment, command.Plan.String()); commentErr != nil {
+			r.Logger.Log(logging.Error, "unable to comment that Atlantis is shutting down: %s", commentErr)
+		}
+		return false
+	}
+	defer r.Drainer.OpDone()
+
+	log := r.Logger.WithHistory(
+		"repository", baseRepo.FullName,
+		"pull-num", strconv.Itoa(pull.Num),
+	)
+	defer r.logPanics(baseRepo, pull.Num, log)
+
+	scope := r.Scope.SubScope("gateway-autoplan")
+	timer := scope.Timer(metrics.ExecutionTimeMetric).Start()
+	defer timer.Stop()
+
+	ctx := &command.Context{
+		User:     user,
+		Log:      log,
+		Scope:    scope,
+		Pull:     pull,
+		HeadRepo: headRepo,
+		Trigger:  command.AutoTrigger,
+	}
+	if !r.validateCtxAndComment(ctx) {
+		return false
+	}
+	if r.DisableAutoplan {
+		return false
+	}
+	err := r.PreWorkflowHooksCommandRunner.RunPreHooks(ctx)
+	if err != nil {
+		ctx.Log.Err("Error running pre-workflow hooks %s. Proceeding with %s command.", err, command.Plan)
```

Comment:
Will our logs get merged with the worker logs? We should figure that out probably and make sure that's not the case.